tests/integration/test_sampling.py

```diff
@@ -297,3 +297,47 @@ def test_extended_sampling_float_special_case_match_star(writer, tracer):
     tracer.configure(sampler=sampler, writer=writer)
     with tracer.trace(name="should_send") as span:
         span.set_tag("tag", 20.1)
+
+
+def test_rate_limiter_on_spans(tracer):
+    """
+    Ensure that the rate limiter is applied to spans
+    """
+    tracer.configure(sampler=DatadogSampler(rate_limit=10))
+    spans = []
+    # Generate 10 spans with the start and finish time in same second
+    for x in range(10):
+        start_time = x / 10
+        span = tracer.trace(name=f"span {start_time}")
+        span.start = start_time
+        span.finish(1 - start_time)
+        spans.append(span)
+    # Generate 11th span in the same second
+    dropped_span = tracer.trace(name=f"span {start_time}")
+    dropped_span.start = 0.8
+    dropped_span.finish(0.9)
+    # Spans are sampled on flush
+    tracer.flush()
+    # Since the rate limiter is set to 10, first ten spans should be kept
+    for span in spans:
```

Comment:
we need to mock time here otherwise we might change time windows in the middle of the test.